Remotion/Data/DomainObjects.UnitTests/Database/TestDomainDBScriptGenerationTest.cs

```diff
@@ -0,0 +1,100 @@
+--
+
+";
+
+    /// <summary>
+    /// Compares the SQL DDL generated from the discovered type definitions to the SQL scripts stored in the Database folder.
+    /// Use the explicit test below to override the stored scripts with the generated ones.
+    /// </summary>
+    [Test]
+    public void GeneratedScriptsMatchStoredScripts ()
+    {
+      foreach (var script in GenerateScripts())
+      {
+        var storageProviderName = script.StorageProviderDefinition.Name;
+        Assert.That(script.SetUpScript, Is.EqualTo(GetStoredScript($"DataDomainObjects_{storageProviderName}_SetupDB.sql")));
+        Assert.That(script.TearDownScript, Is.EqualTo(GetStoredScript($"DataDomainObjects_{storageProviderName}_TearDownDB.sql")));
+      }
+    }
+
+    [Explicit("Run this test to replace the stored scripts with the generated scripts (making the test above green)")]
+    [Test]
+    public void SaveGeneratedScriptsAsNewStoredScripts ()
+    {
+      foreach (var script in GenerateScripts())
+      {
+        var storageProviderName = script.StorageProviderDefinition.Name;
+        SaveStoredScript($"DataDomainObjects_{storageProviderName}_SetupDB.sql", script.SetUpScript);
+        SaveStoredScript($"DataDomainObjects_{storageProviderName}_TearDownDB.sql", script.TearDownScript);
+      }
+    }
+
+    private IEnumerable<Script> GenerateScripts ()
+    {
+      var scriptGenerator = new ScriptGenerator(
+          pd => pd.Factory.CreateSchemaScriptBuilder(pd),
+          new RdbmsStorageEntityDefinitionProvider(),
+          new ScriptToStringConverter());
+
+      var typeDefinitions = MappingConfiguration.Current.GetTypeDefinitions()
+          .Where(e => e.StorageEntityDefinition.StorageProviderID is c_testDomainProviderID or TableInheritanceMappingTest.TableInheritanceTestDomainProviderID)
+          .Where(e => !Attribute.IsDefined(e.ClassType, typeof(ExcludeFromTestDomainDBAttribute)));
+
+      return scriptGenerator.GetScripts(typeDefinitions)
+          .Select(e => new Script(e.StorageProviderDefinition, FixGeneratedScript(e.SetUpScript), FixGeneratedScript(e.TearDownScript)));
+    }
+
+    private void SaveStoredScript (string name, string newContent) // todo dev suffix
+    {
+      var filePath = Path.GetFullPath(Path.Combine(TestContext.CurrentContext.TestDirectory, "../../../Database", name));
+      Assert.True(File.Exists(filePath), "Could not find stored SQL script '{0}'.", filePath);
+
+      File.WriteAllText(filePath, newContent);
+    }
+
+    private string FixGeneratedScript (string script)
+    {
+      return c_scriptPrefix + script.Replace("USE TestDomain", "USE DBPrefix_TestDomain");
+    }
+
+    private string GetStoredScript (string name)
+    {
+      var filePath = Path.GetFullPath(Path.Combine(TestContext.CurrentContext.TestDirectory, $"Database/{name}"));
+      Assert.True(File.Exists(filePath), "Could not find stored SQL script '{0}'.", filePath);
```

Comment:
```suggestion
      Assert.That(filePath, Does.Exist, "Could not find stored SQL script.");
```

Reply:
TIL that there is a `Does.Exist`

Reply:
:) with NUnit, we may assume that there's an api to deal with collection-stuff, number-compare, String-contents, and FileSystem stuff. I usually try to assume someone already did this when I need a check and try to disprove it with google / docs :)